### PR TITLE
Add OS compatibility matrix for basic keycodes

### PR DESCRIPTION
**Pull request by fauxpark**
## Description

Documenting which keycodes work on the big three, which don't, and which have weird/nonstandard behaviour (mostly on macOS). Would appreciate confirmation/testing of the unknown ones :)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: docs/keycodes.md [hunk 1]
```diff
@@ -8,205 +8,210 @@ This is a reference only. Each group of keys links to the page documenting their
...
+|`KC_F24`               |                              |F24                                            |?      |?            |?    |
+|`KC_EXECUTE`           |`KC_EXEC`                     |Execute                                        |?      |?            |?    |
+|`KC_HELP`              |                              |Help                                           |?      |?            |?    |
+|`KC_MENU`              |                              |Menu                                           |?      |?            |?    |
+|`KC_SELECT`            |`KC_SLCT`                     |Select                                         |?      |?            |?    |
+|`KC_STOP`              |                              |Stop                                           |?      |?            |?    |
+|`KC_AGAIN`             |`KC_AGIN`                     |Again                                          |?      |?            |?    |
+|`KC_UNDO`              |                              |Undo                                           |?      |?            |?    |
```
Comment (Erovia) [inline docs/keycodes.md:133]:
```suggestion
|`KC_POWER`             |                              |System Power                                   |?      |✔<sup>2</sup>|✔    |
|`KC_KP_EQUAL`          |`KC_PEQL`                     |Keypad `=`                                     |?      |?            |✔    |
|`KC_F13`               |                              |F13                                            |?      |?            |✔    |
|`KC_F14`               |                              |F14                                            |?      |?            |✔    |
|`KC_F15`               |                              |F15                                            |?      |?            |✔    |
|`KC_F16`               |                              |F16                                            |?      |?            |✔    |
|`KC_F17`               |                              |F17                                            |?      |?            |✔    |
|`KC_F18`               |                              |F18                                            |?      |?            |✔    |
|`KC_F19`               |                              |F19                                            |?      |?            |✔    |
|`KC_F20`               |                              |F20                                            |?      |?            |✔    |
|`KC_F21`               |                              |F21                                            |?      |?            |✔    |
|`KC_F22`               |                              |F22                                            |?      |?            |✔    |
|`KC_F23`               |                              |F23                                            |?      |?            |✔    |
|`KC_F24`               |                              |F24                                            |?      |?            |✔    |
|`KC_EXECUTE`           |`KC_EXEC`                     |Execute                                        |?      |?            |✔    |
|`KC_HELP`              |                              |Help                                           |?      |?            |✔    |
|`KC_MENU`              |                              |Menu                                           |?      |?            |✔    |
|`KC_SELECT`            |`KC_SLCT`                     |Select                                         |?      |?            |✔    |
|`KC_STOP`              |                              |Stop                                           |?      |?            |✔    |
|`KC_AGAIN`             |`KC_AGIN`                     |Again                                          |?      |?            |✔    |
|`KC_UNDO`              |                              |Undo                                           |?      |?            |✔    |
```

File: docs/keycodes.md [hunk 2]
```diff
@@ -8,205 +8,210 @@ This is a reference only. Each group of keys links to the page documenting their
...
+|`KC_FIND`              |                              |Find                                           |?      |?            |?    |
+|`KC__MUTE`             |                              |Mute                                           |?      |✔            |?    |
+|`KC__VOLUP`            |                              |Volume Up                                      |?      |✔            |?    |
+|`KC__VOLDOWN`          |                              |Volume Down                                    |?      |✔            |?    |
+|`KC_LOCKING_CAPS`      |`KC_LCAP`                     |Locking Caps Lock                              |?      |?            |?    |
+|`KC_LOCKING_NUM`       |`KC_LNUM`                     |Locking Num Lock                               |?      |?            |?    |
+|`KC_LOCKING_SCROLL`    |`KC_LSCR`                     |Locking Scroll Lock                            |?      |?            |?    |
+|`KC_KP_COMMA`          |`KC_PCMM`                     |Keypad `,`                                     |?      |?            |?    |
```
Comment (Erovia) [inline docs/keycodes.md:144]:
```suggestion
|`KC_FIND`              |                              |Find                                           |?      |?            |✔    |
|`KC__MUTE`             |                              |Mute                                           |?      |✔            |✔    |
|`KC__VOLUP`            |                              |Volume Up                                      |?      |✔            |✔    |
|`KC__VOLDOWN`          |                              |Volume Down                                    |?      |✔            |✔    |
|`KC_LOCKING_CAPS`      |`KC_LCAP`                     |Locking Caps Lock                              |?      |?            |✔    |
|`KC_LOCKING_NUM`       |`KC_LNUM`                     |Locking Num Lock                               |?      |?            |✔    |
|`KC_LOCKING_SCROLL`    |`KC_LSCR`                     |Locking Scroll Lock                            |?      |?            |✔    |
|`KC_KP_COMMA`          |`KC_PCMM`                     |Keypad `,`                                     |?      |?            |✔    |
```

File: docs/keycodes.md [hunk 3]
```diff
@@ -8,205 +8,210 @@ This is a reference only. Each group of keys links to the page documenting their
...
+|`KC_KP_EQUAL_AS400`    |                              |Keypad `=` on AS/400 keyboards                 |?      |?            |?    |
+|`KC_INT1`              |`KC_RO`                       |JIS `\` and `_`                                |       |             |     |
+|`KC_INT2`              |`KC_KANA`                     |JIS Katakana/Hiragana                          |       |             |     |
+|`KC_INT3`              |`KC_JYEN`                     |JIS `¥` and `\|`                               |       |             |     |
+|`KC_INT4`              |`KC_HENK`                     |JIS Henkan                                     |       |             |     |
+|`KC_INT5`              |`KC_MHEN`                     |JIS Muhenkan                                   |       |             |     |
+|`KC_INT6`              |                              |JIS Numpad `,`                                 |       |             |     |
```
Comment (Erovia) [inline docs/keycodes.md:151]:
```suggestion
|`KC_INT1`              |`KC_RO`                       |JIS `\` and `_`                                |       |             |✔    |
|`KC_INT2`              |`KC_KANA`                     |JIS Katakana/Hiragana                          |       |             |✔    |
|`KC_INT3`              |`KC_JYEN`                     |JIS `¥` and `\|`                               |       |             |✔    |
|`KC_INT4`              |`KC_HENK`                     |JIS Henkan                                     |       |             |✔    |
|`KC_INT5`              |`KC_MHEN`                     |JIS Muhenkan                                   |       |             |✔    |
|`KC_INT6`              |                              |JIS Numpad `,`                                 |       |             |✔    |
```

File: docs/keycodes.md [hunk 4]
```diff
@@ -8,205 +8,210 @@ This is a reference only. Each group of keys links to the page documenting their
...
+|`KC_INT7`              |                              |International 7                                |       |             |     |
+|`KC_INT8`              |                              |International 8                                |       |             |     |
+|`KC_INT9`              |                              |International 9                                |       |             |     |
+|`KC_LANG1`             |`KC_HAEN`                     |Hangul/English                                 |       |             |     |
+|`KC_LANG2`             |`KC_HANJ`                     |Hanja                                          |       |             |     |
+|`KC_LANG3`             |                              |JIS Katakana                                   |       |             |     |
+|`KC_LANG4`             |                              |JIS Hiragana                                   |       |             |     |
+|`KC_LANG5`             |                              |JIS Zenkaku/Hankaku                            |       |             |     |
```
Comment (Erovia) [inline docs/keycodes.md:159]:
```suggestion
|`KC_LANG1`             |`KC_HAEN`                     |Hangul/English                                 |       |             |✔    |
|`KC_LANG2`             |`KC_HANJ`                     |Hanja                                          |       |             |✔    |
|`KC_LANG3`             |                              |JIS Katakana                                   |       |             |✔    |
|`KC_LANG4`             |                              |JIS Hiragana                                   |       |             |✔    |
|`KC_LANG5`             |                              |JIS Zenkaku/Hankaku                            |       |             |✔    |
```

File: docs/keycodes.md [hunk 5]
```diff
@@ -8,205 +8,210 @@ This is a reference only. Each group of keys links to the page documenting their
...
+|`KC_LANG6`             |                              |Language 6                                     |       |             |     |
+|`KC_LANG7`             |                              |Language 7                                     |       |             |     |
+|`KC_LANG8`             |                              |Language 8                                     |       |             |     |
+|`KC_LANG9`             |                              |Language 9                                     |       |             |     |
+|`KC_ALT_ERASE`         |`KC_ERAS`                     |Alternate Erase                                |       |             |     |
+|`KC_SYSREQ`            |                              |SysReq/Attention                               |       |             |     |
+|`KC_CANCEL`            |                              |Cancel                                         |       |             |     |
+|`KC_CLEAR`             |`KC_CLR`                      |Clear                                          |       |             |     |
```
Comment (Erovia) [inline docs/keycodes.md:167]:
```suggestion
|`KC_CLEAR`             |`KC_CLR`                      |Clear                                          |       |             |✔    |
```

File: docs/keycodes.md [hunk 6]
```diff
@@ -8,205 +8,210 @@ This is a reference only. Each group of keys links to the page documenting their
...
+|`KC_SYSTEM_SLEEP`      |`KC_SLEP`                     |System Sleep                                   |✔      |✔<sup>2</sup>|✔    |
+|`KC_SYSTEM_WAKE`       |`KC_WAKE`                     |System Wake                                    |?      |?            |?    |
+|`KC_AUDIO_MUTE`        |`KC_MUTE`                     |Mute                                           |✔      |✔            |✔    |
+|`KC_AUDIO_VOL_UP`      |`KC_VOLU`                     |Volume Up                                      |✔      |✔<sup>3</sup>|✔    |
+|`KC_AUDIO_VOL_DOWN`    |`KC_VOLD`                     |Volume Down                                    |✔      |✔<sup>3</sup>|✔    |
+|`KC_MEDIA_NEXT_TRACK`  |`KC_MNXT`                     |Next Track                                     |✔      |✔<sup>4</sup>|✔    |
+|`KC_MEDIA_PREV_TRACK`  |`KC_MPRV`                     |Previous Track                                 |✔      |✔<sup>4</sup>|✔    |
+|`KC_MEDIA_STOP`        |`KC_MSTP`                     |Stop Track                                     |✔      |?            |?    |
```
Comment (Erovia) [inline docs/keycodes.md:192]:
```suggestion
|`KC_MEDIA_STOP`        |`KC_MSTP`                     |Stop Track                                     |✔      |?            |✔    |
```

File: docs/keycodes.md [hunk 7]
```diff
@@ -8,205 +8,210 @@ This is a reference only. Each group of keys links to the page documenting their
...
+|`KC_MEDIA_PLAY_PAUSE`  |`KC_MPLY`                     |Play/Pause Track                               |✔      |✔            |✔    |
+|`KC_MEDIA_SELECT`      |`KC_MSEL`                     |Launch Media Player                            |✔      |?            |?    |
```
Comment (Erovia) [inline docs/keycodes.md:194]:
```suggestion
|`KC_MEDIA_SELECT`      |`KC_MSEL`                     |Launch Media Player                            |✔      |?            |✔    |
```

File: docs/keycodes.md [hunk 8]
```diff
@@ -8,205 +8,210 @@ This is a reference only. Each group of keys links to the page documenting their
...
+|`KC_WWW_HOME`          |`KC_WHOM`                     |Browser Home                                   |✔      |             |?    |
+|`KC_WWW_BACK`          |`KC_WBAK`                     |Browser Back                                   |✔      |             |?    |
+|`KC_WWW_FORWARD`       |`KC_WFWD`                     |Browser Forward                                |✔      |             |?    |
+|`KC_WWW_STOP`          |`KC_WSTP`                     |Browser Stop                                   |✔      |             |?    |
+|`KC_WWW_REFRESH`       |`KC_WREF`                     |Browser Refresh                                |✔      |             |?    |
+|`KC_WWW_FAVORITES`     |`KC_WFAV`                     |Browser Favorites                              |✔      |             |?    |
+|`KC_MEDIA_FAST_FORWARD`|`KC_MFFD`                     |Next Track                                     |?      |✔<sup>4</sup>|?    |
+|`KC_MEDIA_REWIND`      |`KC_MRWD`                     |Previous Track                                 |?      |✔<sup>4</sup>|?    |
```
Comment (Erovia) [inline docs/keycodes.md:207]:
```suggestion
|`KC_MAIL`              |                              |Launch Mail                                    |✔      |             |✔    |
|`KC_CALCULATOR`        |`KC_CALC`                     |Launch Calculator                              |✔      |             |✔    |
|`KC_MY_COMPUTER`       |`KC_MYCM`                     |Launch My Computer                             |✔      |             |✔    |
|`KC_WWW_SEARCH`        |`KC_WSCH`                     |Browser Search                                 |✔      |             |✔    |
|`KC_WWW_HOME`          |`KC_WHOM`                     |Browser Home                                   |✔      |             |✔    |
|`KC_WWW_BACK`          |`KC_WBAK`                     |Browser Back                                   |✔      |             |✔    |
|`KC_WWW_FORWARD`       |`KC_WFWD`                     |Browser Forward                                |✔      |             |✔    |
|`KC_WWW_STOP`          |`KC_WSTP`                     |Browser Stop                                   |✔      |             |✔    |
|`KC_WWW_REFRESH`       |`KC_WREF`                     |Browser Refresh                                |✔      |             |✔    |
|`KC_WWW_FAVORITES`     |`KC_WFAV`                     |Browser Favorites                              |✔      |             |✔    |
|`KC_MEDIA_FAST_FORWARD`|`KC_MFFD`                     |Next Track                                     |?      |✔<sup>4</sup>|✔    |
|`KC_MEDIA_REWIND`      |`KC_MRWD`                     |Previous Track                                 |?      |✔<sup>4</sup>|✔    |
```

File: docs/keycodes.md [hunk 9]
```diff
@@ -8,205 +8,210 @@ This is a reference only. Each group of keys links to the page documenting their
...
+|`KC__MUTE`             |                              |Mute                                           |?      |✔            |?    |
+|`KC__VOLUP`            |                              |Volume Up                                      |?      |✔            |?    |
+|`KC__VOLDOWN`          |                              |Volume Down                                    |?      |✔            |?    |
+|`KC_LOCKING_CAPS`      |`KC_LCAP`                     |Locking Caps Lock                              |?      |?            |?    |
+|`KC_LOCKING_NUM`       |`KC_LNUM`                     |Locking Num Lock                               |?      |?            |?    |
+|`KC_LOCKING_SCROLL`    |`KC_LSCR`                     |Locking Scroll Lock                            |?      |?            |?    |
+|`KC_KP_COMMA`          |`KC_PCMM`                     |Keypad `,`                                     |?      |?            |?    |
+|`KC_KP_EQUAL_AS400`    |                              |Keypad `=` on AS/400 keyboards                 |?      |?            |?    |
```
Comment (Erovia) [inline docs/keycodes.md:145]:
```suggestion
|`KC_KP_EQUAL_AS400`    |                              |Keypad `=` on AS/400 keyboards                 |?      |?            |     |
```

File: docs/keycodes.md [hunk 10]
```diff
@@ -8,205 +8,210 @@ This is a reference only. Each group of keys links to the page documenting their
...
+|`KC_AUDIO_VOL_UP`      |`KC_VOLU`                     |Volume Up                                      |✔      |✔<sup>3</sup>|✔    |
+|`KC_AUDIO_VOL_DOWN`    |`KC_VOLD`                     |Volume Down                                    |✔      |✔<sup>3</sup>|✔    |
+|`KC_MEDIA_NEXT_TRACK`  |`KC_MNXT`                     |Next Track                                     |✔      |✔<sup>4</sup>|✔    |
+|`KC_MEDIA_PREV_TRACK`  |`KC_MPRV`                     |Previous Track                                 |✔      |✔<sup>4</sup>|✔    |
+|`KC_MEDIA_STOP`        |`KC_MSTP`                     |Stop Track                                     |✔      |?            |?    |
+|`KC_MEDIA_PLAY_PAUSE`  |`KC_MPLY`                     |Play/Pause Track                               |✔      |✔            |✔    |
+|`KC_MEDIA_SELECT`      |`KC_MSEL`                     |Launch Media Player                            |✔      |?            |?    |
+|`KC_MEDIA_EJECT`       |`KC_EJCT`                     |Eject                                          |?      |✔            |?    |
```
Comment (Erovia) [inline docs/keycodes.md:195]:
```suggestion
|`KC_MEDIA_EJECT`       |`KC_EJCT`                     |Eject                                          |?      |✔            |     |
```

File: docs/keycodes.md [hunk 11]
```diff
@@ -8,205 +8,210 @@ This is a reference only. Each group of keys links to the page documenting their
...
+|`KC_LGUI`              |`KC_LCMD`, `KC_LWIN`          |Left GUI (Windows/Command/Meta key)            |✔      |✔            |✔    |
+|`KC_RCTRL`             |`KC_RCTL`                     |Right Control                                  |✔      |✔            |✔    |
+|`KC_RSHIFT`            |`KC_RSFT`                     |Right Shift                                    |✔      |✔            |✔    |
+|`KC_RALT`              |`KC_ROPT`, `KC_ALGR`          |Right Alt (Option/AltGr)                       |✔      |✔            |✔    |
+|`KC_RGUI`              |`KC_RCMD`, `KC_RWIN`          |Right GUI (Windows/Command/Meta key)           |✔      |✔            |✔    |
+|`KC_SYSTEM_POWER`      |`KC_PWR`                      |System Power Down                              |✔      |✔<sup>2</sup>|✔    |
+|`KC_SYSTEM_SLEEP`      |`KC_SLEP`                     |System Sleep                                   |✔      |✔<sup>2</sup>|✔    |
+|`KC_SYSTEM_WAKE`       |`KC_WAKE`                     |System Wake                                    |?      |?            |?    |
```
Comment (Erovia) [inline docs/keycodes.md:186]:
```suggestion
|`KC_SYSTEM_WAKE`       |`KC_WAKE`                     |System Wake                                    |?      |?            |✔    |
```

File: docs/keycodes.md [hunk 12]
```diff
@@ -8,205 +8,210 @@ This is a reference only. Each group of keys links to the page documenting their
...
+|`KC_LANG4`             |                              |JIS Hiragana                                   |       |             |     |
+|`KC_LANG5`             |                              |JIS Zenkaku/Hankaku                            |       |             |     |
+|`KC_LANG6`             |                              |Language 6                                     |       |             |     |
+|`KC_LANG7`             |                              |Language 7                                     |       |             |     |
+|`KC_LANG8`             |                              |Language 8                                     |       |             |     |
+|`KC_LANG9`             |                              |Language 9                                     |       |             |     |
+|`KC_ALT_ERASE`         |`KC_ERAS`                     |Alternate Erase                                |       |             |     |
+|`KC_SYSREQ`            |                              |SysReq/Attention                               |       |             |     |
```
Comment (Erovia) [inline docs/keycodes.md:165]:
```suggestion
|`KC_SYSREQ`            |                              |SysReq/Attention                               |       |             |✔    |
```

File: docs/keycodes.md [hunk 13]
```diff
@@ -8,205 +8,210 @@ This is a reference only. Each group of keys links to the page documenting their
...
+|`KC_WWW_FAVORITES`     |`KC_WFAV`                     |Browser Favorites                              |✔      |             |?    |
+|`KC_MEDIA_FAST_FORWARD`|`KC_MFFD`                     |Next Track                                     |?      |✔<sup>4</sup>|?    |
+|`KC_MEDIA_REWIND`      |`KC_MRWD`                     |Previous Track                                 |?      |✔<sup>4</sup>|?    |
+|`KC_BRIGHTNESS_UP`     |`KC_BRIU`                     |Brightness Up                                  |✔      |✔            |✔    |
+|`KC_BRIGHTNESS_DOWN`   |`KC_BRID`                     |Brightness Down                                |✔      |✔            |✔    |
+
+<sup>1. Recognized as F13-F15.</sup><br/>
+<sup>2. Must be held for about three seconds, and will display a prompt instead.</sup><br/>
+<sup>3. Holding Shift+Option allows for finer control of volume level.</sup><br/>
+<sup>4. Skips the entire track in iTunes when tapped, seeks within the current track when held.</sup><br/>
```
Comment (Erovia) [inline docs/keycodes.md:214]:
```suggestion
<sup>4. Skips the entire track in iTunes when tapped, seeks within the current track when held.</sup><br/>
<sup>5. Tested with Linux 5.5.15 and xorg-x11-server 1.20.6</sup>
```

**Comment by fauxpark** [comment]
I think I would prefer if there were a separate symbol for keycodes that are only recognised, with the tick denoting ones that actually do something useful.
The issue is that if we simply give all keycodes that are *at least* recognised a tick, people will be confused when they try those out, expecting them to have some kind of effect.
For example, the context menu key `KC_APP` is recognised on macOS, and is even passed to the browser as keycode.info picks it up perfectly fine. But it doesn't open the right-click menu like it does on Windows and (I think) Linux.

**Comment by spidey3** [comment]
> I think I would prefer if there were a separate symbol for keycodes that are only recognised, with the tick denoting ones that actually do something useful.
I also think this would be a good idea. 
At the moment, in my personal notes, I only recorded something as worthy of a ✔ if it actually does something. The exception to this is the F<n> keys, because those are always just available to be bound to some action.

**Comment by Erovia** [comment]
I'm still not sure how that approach could be applied to Linux. Both Mac and Windows has it's desktop environment so checking if something does have an effect is easy.
On Linux every Desktop Environment / Windows Manager can do have different defaults. (and they usually have)  

Example: the mentioned `KC_APP` opens the context menu for me on GNOME3 (out of the box), but I'm pretty sure it will do nothing on `i3wm`.

tl;dr: Help me define "has effect" on Linux. Closest thing probably is to add a note that this was done on this and this DE/WM.

**Comment by spidey3** [comment]
> tl;dr: Help me define "has effect" on Linux. Closest thing probably is to add a note that this was done on this and this DE/WM.

The approach I took with MacOS is "has an effect by default OR can be bound to an action with no extra software".  For Linux that would be "is bound to an action _in some common desktop environments_ OR can easily be bound...".  Then add a big fat caveat to the documentation explaining that assumption.

**Comment by Erovia** [comment]
I've yet to encounter a DE/WM that does not have some kind of shortcuts feature.
As for the caveat, my suggestion was/is this: https://github.com/qmk/qmk_firmware/pull/8557#discussion_r407493561

**Comment by spidey3** [comment]
FYI, I have confirmed that:

- the following work on MacOS, and should have a ✔:
```KC_KP_EQUAL, KC_F13…KC_F19, KC_LOCKING_CAPS, KC_LOCKING_NUM, KC_SYSTEM_WAKE```
- this works, but just has the effect of holding down `KC_F14`:
```KC_LOCKING_SCROLL```
- the following do NOT work in MacOS:
```KC_F20…KC_F24, KC_EXECUTE, KC_HELP, KC_MENU, KC_SELECT, KC_STOP, KC_AGAIN, KC_UNDO, KC_CUT, KC_COPY, KC_PASTE, KC_FIND, KC_KP_COMMA, KC_KP_EQUAL_AS400, KC_MEDIA_STOP, KC_MEDIA_SELECT```

**Comment by spidey3** [comment]
More confirmations, this time on Windows:
- The following work on Windows, and should have a ✔:
```KC_F13…KC_F24```
- the following do NOT work in Windows:
```KC_EXECUTE, KC_HELP, KC_MENU, KC_SELECT, KC_STOP, KC_AGAIN, KC_UNDO, KC_CUT, KC_COPY, KC_PASTE, KC_FIND```

**Comment by spidey3** [comment]
A few more on Windows:
- KC_KP_EQUAL_AS400 - no
- KC_WAKE - it wakes up a sleeping computer, but so does any other key...
- KC_MEDIA_STOP - no (at least not in iTunes or in Windows Media Center)
- KC_MEDIA_SELECT - yes (launches the default media player)

**Comment by spidey3** [comment]
@fauxpark - is there any way that I can help move this forward?

**Comment by fauxpark** [comment]
@Erovia @spidey3 I've applied your findings, let me know if I missed anything.
